category_encoders/rankhot.py

```diff
@@ -0,0 +1,263 @@
+           options are 'error', 'ignore'. The default is 'ignore'. If unknown variable exists then it is represented as 0 array. Else error message
+           is displayed
+       handle_missing: str
+           options are 'error' and 'value'. The default is 'value'. Missing value also considered as unknown value in the final data set.
+
+       Example
+       -------
+       >>> from category_encoders import *
+       >>> import pandas as pd
+       >>> from sklearn.datasets import load_boston
+       >>> bunch = load_boston()
+       >>> y = bunch.target
+       >>> X = pd.DataFrame(bunch.data, columns=bunch.feature_names)
+       >>> enc = RankHotEncoder(cols=['CHAS'], handle_unknown='ignore').fit(X)
+       >>> numeric_dataset = enc.transform(X)
+       >>> print(numeric_dataset.info())
+    """
+
+    """
+    <class 'pandas.core.frame.DataFrame'>
+    RangeIndex: 506 entries, 0 to 505
+    Data columns (total 14 columns):
+    CRIM       506 non-null float64
+    ZN         506 non-null float64
+    INDUS      506 non-null float64
+    CHAS_1     506 non-null int32
+    CHAS_2     506 non-null int32
+    NOX        506 non-null float64
+    RM         506 non-null float64
+    AGE        506 non-null float64
+    DIS        506 non-null float64
+    RAD        506 non-null float64
+    TAX        506 non-null float64
+    PTRATIO    506 non-null float64
+    B          506 non-null float64
+    LSTAT      506 non-null float64   
+    """
+    def __init__(self, verbose = 0, cols = None, drop_invariant = False, handle_missing='value', handle_unknown='ignore', use_cat_names = None):
+        self.cols = cols
+        self.ordinal_encoder = None
+        self.handle_unknown = handle_unknown
+        self.handle_missing = handle_missing
+        self._dim = None
+        self.verbose = verbose
+        self.mapping = None
+        self.feature_names = None
+        self.drop_invariant = drop_invariant
+        self.use_cat_names = use_cat_names
+
+    def fit(self, X):
+
+        self._dim = X.shape[0]
+        self.feature_names = X.columns
+
+        self.ordinal_encoder = OrdinalEncoder(
+            verbose=self.verbose,
+            cols=self.cols,
+            handle_unknown='value',
+            handle_missing='value'
+        )
+
+        self.ordinal_encoder = self.ordinal_encoder.fit(X)
+
+        self.mapping = self.generate_mapping()
+
+        self.ordinal_encoder.category_mapping[0]['mapping'] =  self.ordinal_encoder.category_mapping[0]['mapping'].drop(labels = [float('nan')])
+
+        X_temp = self.transform(X, override_return_df=True)
+        self.feature_names = list(X_temp.columns)
+
+        if self.drop_invariant:
+            self.drop_cols = []
+            generated_cols = util.get_generated_cols(X, X_temp, self.cols)
+            self.drop_cols = [x for x in generated_cols if X_temp[x].var() <= 10e-5]
+            try:
+                [self.feature_names.remove(x) for x in self.drop_cols]
```

Comment:
```suggestion
                for x in self.drop_cols:
                    self.feature_names.remove(x)
```

don't use list comprehensions for side effects.